### ISSUE TITLE
Keep fragment on redirects

### ISSUE THREAD

**Issue by annulen** [annulen/webkit#623]
```diff
diff --git a/Source/WebCore/platform/network/qt/QNetworkReplyHandler.cpp b/Source/WebCore/platform/network/qt/QNetworkReplyHandler.cpp
index cd096bfe7f0..1b60c513175 100644
--- a/Source/WebCore/platform/network/qt/QNetworkReplyHandler.cpp
+++ b/Source/WebCore/platform/network/qt/QNetworkReplyHandler.cpp
@@ -647,7 +647,10 @@ void QNetworkReplyHandler::redirect(ResourceResponse& response, const QUrl& redi
 {
     ASSERT(!m_queue.deferSignals());
 
-    QUrl newUrl = m_replyWrapper->reply()->url().resolved(redirection);
+    QUrl currentUrl = m_replyWrapper->reply()->url();
+    QUrl newUrl = currentUrl.resolved(redirection);
+    if (currentUrl.hasFragment())
+        newUrl.setFragment(currentUrl.fragment());
 
     ResourceHandleClient* client = m_resourceHandle->client();
     ASSERT(client);
```

TODO: run layout tests, find if we have some layout test for it

**Comment by annulen** [comment]
Actually this patch is broken, as it overrides fragment in newUrl when it's present in `Location` header, correct condition is `if (!newUrl.hasFragment() && currentUrl.hasFragment())`

**Comment by annulen** [comment]
Fixed in f647a7c5712e1bc078e4ec837b50def0b5b9e362
